lib/Interfaces.tsx

```diff
@@ -1,273 +1,73 @@
 import {FieldValue, serverTimestamp, Timestamp} from "@firebase/firestore";
 
-export interface Budget {
-    id: string;
-    category_name: string;
-    amount: number;
-    spent?: number; // adding this as a computed value
-    is_monthly: boolean;
-    is_yearly: boolean;
-    icon?: string;
+export type User = {
+    uid: string;
+    email: string;
+    photoURL: string | null;
 }
 
-export class BudgetClass {
-    id = "";
-    category_name = "";
-    amount = 0;
-    is_monthly = false;
-    is_yearly = false;
-
-
-    constructor(category_name: string, amount: number, is_monthly?: boolean, is_yearly?: boolean) {
-        this.id = this.generateBudgetId(category_name)
-        this.category_name = category_name;
-        this.amount = amount;
-        this.is_monthly = is_monthly ?? true;
-        this.is_yearly = is_yearly ?? false;
-
-    }
-
-    generateBudgetId(category_name: string): string {
-        // Get current timestamp
-        const timestamp = Date.now();
-
-        // Construct the ID
-        return `${category_name}_${timestamp}`;
-    }
-
-    toObject() {
-        return {
-            id: this.id,
-            category_name: this.category_name,
-            amount: this.amount,
-            is_monthly: this.is_monthly,
-            is_yearly: this.is_yearly
+export type Month = {
+    month: number;
+    year: number;
+    totals: {
+        monthTotal: number;
+        budgetTotals: {
+            [budgetID: string]: number;
         }
     }
 }
 
-export interface Category {
-    id: string;
-    month: string;
-    budget: number;
-    category_name: string;
-    year: number;
-    spent: number;
-    expenses: string[];
-    iconName: string;
+export type Budget = {
+    budgetID: string;
```

Comment:
Sorry if I missed this somewhere, but what does budgetID represent? Is it a uniquely generated string for each Budget? Can you give an example?